File: scripts/tasks/bootstrap.ts
import { commandExists, logStep, runOrThrow } from "../common";

const REQUIRED_TOOLS = ["bun", "git", "curl"];

export function bootstrap(): void {
  logStep("Checking prerequisites");
  for (const tool of REQUIRED_TOOLS) {
    if (!commandExists(tool)) {
      throw new Error(`Missing required tool: ${tool}`);
    }
    console.log(`ok: ${tool}`);
  }

  logStep("Installing dependencies with Bun");
  runOrThrow(["bun", "install"]);

  logStep("Installing managed project dependencies");
  runOrThrow(["bun", "run", "scry:projects:install"]);

  logStep("Preparing local storage defaults");
  runOrThrow(["bun", "run", "scry:setup:storage"]);

  logStep("Bootstrap complete");
  const bunVersion = runOrThrow(["bun", "--version"], { quiet: true });
  console.log(`bun: ${bunVersion}`);
}

File: scripts/tasks/bootstrap.py
"""Bootstrap grimoire: check prerequisites, install deps."""

from __future__ import annotations

from scripts.common import command_exists, log_step, run_or_throw

REQUIRED_TOOLS = ["bun", "git", "curl"]


def bootstrap() -> None:
    log_step("Checking prerequisites")
    for tool in REQUIRED_TOOLS:
        if not command_exists(tool):
            raise RuntimeError(f"Missing required tool: {tool}")
        print(f"ok: {tool}")

    log_step("Installing dependencies with Bun")
    run_or_throw(["bun", "install"])

    log_step("Installing managed project dependencies")
    run_or_throw(["uv", "run", "python", "-m", "scripts", "projects:install"])

    log_step("Bootstrap complete")
    bun_version = run_or_throw(["bun", "--version"], quiet=True)
    print(f"bun: {bun_version}")
